src/core/core.cpp

```diff
@@ -766,59 +764,61 @@ wayfire_view wf::compositor_core_impl_t::find_view(const std::string& id)
 
 pid_t wf::compositor_core_impl_t::run(std::string command)
 {
-    static constexpr size_t READ_END  = 0;
-    static constexpr size_t WRITE_END = 1;
     pid_t pid;
-    int pipe_fd[2];
-    pipe2(pipe_fd, O_CLOEXEC);
+    int daemonized, status, sp[2];
+    struct pollfd pfd;
+    int TIMEOUT = 500;
+
+    if (socketpair(AF_UNIX, SOCK_STREAM, PF_UNSPEC, sp) == -1)
+    {
+      return -1;
+    }
 
-    /* The following is a "hack" for disowning the child processes,
```

Comment:
I think the comment should stay. We still do a double fork (even though daemonize does it for us), and I think that the purpose of this may not be clear to all readers.

Reply:
readded